crates/ra_ide/src/diagnostics.rs

```diff
@@ -715,4 +715,34 @@ fn main() {
             check_struct_shorthand_initialization,
         );
     }
+
+    #[test]
+    fn test_bad_macro_stackover() {
```

Comment:
could this test be in `hir_ty` instead?

Reply:
Sure. 

Originally I found out the bug is triggered in diagnostics stage so I put the test in there but hir_ty is a better place to put.